src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -231,12 +232,30 @@ private boolean areTotalsEqual(InvoiceLine invoiceLine1, InvoiceLine invoiceLine
       }));
   }
 
+  /**
+   * Deletes Invoice Line and update Invoice if deletion is allowed
+   * 1. Get invoiceLine by id
+   * 2. Verify if user has permission to delete invoiceLine based on acquisitions units
+   * 3. If user has permission to delete then delete invoiceLine
+   * 4. Update corresponding Invoice
+   * @param id invoiceLine id to be deleted
+   */
   public CompletableFuture<Void> deleteInvoiceLine(String id) {
-    return getInvoiceLine(id)
+    return getInvoiceLine(id).thenCompose(this::verifyDeleteAllowed)
       .thenCompose(line -> handleDeleteRequest(resourceByIdPath(INVOICE_LINES, id, lang), httpClient, ctx, okapiHeaders, logger)
         .thenRun(() -> updateInvoiceAndLinesAsync(line.getInvoiceId())));
   }
 
+  private CompletableFuture<InvoiceLine> verifyDeleteAllowed(InvoiceLine invoiceLine) {
+    return getInvoiceRecord(invoiceLine.getInvoiceId())
+      .thenCompose(invoice -> protectionHelper.isOperationRestricted(invoice.getAcqUnitIds(), DELETE))
+      .thenApply(aVoid -> invoiceLine);
+  }
+
+  public CompletableFuture<Invoice> getInvoiceRecord(String id) {
```

Comment:
This method does nothing but call `HelperUtil#getInvoiceById`...

Reply:
In general, it is useful for code readability, but in this case it is worth replacing other mentions of `getInvoiceById`.